### PR TITLE
Add details to Histogram API

### PR DESCRIPTION
**Pull request by reyang**
Follow up #1617.

## Changes

Added details to the Histogram instrument. In #1617 these were left as TODO.

### CODE REVIEW

File: specification/metrics/new_api.md [hunk 1]
```diff
@@ -461,13 +461,77 @@ Example uses for `Histogram`:
...
 
 ##### Record
 
-TODO
+Updates the statistics with the specified amount.
+
+This API SHOULD NOT return a value (it MAY return a dummy value if required by
+certain programming languages or systems, for example `null`, `undefined`).
+
+Required parameters:
```
Comment (cijothomas) [inline specification/metrics/new_api.md:511]: nit:

```suggestion
Parameters:
```
as one parameter is optional.

File: specification/metrics/new_api.md [hunk 2]
```diff
@@ -461,13 +461,77 @@ Example uses for `Histogram`:
...
+
+Here are some examples that individual language client might consider:
+
+```python
+# Python
+
+http_server_duration = meter.create_histogram(
+    name="http.server.duration",
+    description="measures the duration of the inbound HTTP request",
+    uint="milliseconds",
```
Comment (cijothomas) [inline specification/metrics/new_api.md:487]:
```suggestion
    unit="milliseconds",
```

File: specification/metrics/new_api.md [hunk 3]
```diff
@@ -461,13 +461,77 @@ Example uses for `Histogram`:
...
+    value_type=float)
+```
+
+```csharp
+// C#
+
+var httpServerDuration = meter.CreateHistogram<double>(
+    "http.server.duration",
+    description: "measures the duration of the inbound HTTP request",
+    uint: "milliseconds",
```
Comment (cijothomas) [inline specification/metrics/new_api.md:497]:
```suggestion
    unit: "milliseconds",
```

File: specification/metrics/new_api.md [hunk 4]
```diff
@@ -461,13 +461,77 @@ Example uses for `Histogram`:
...
+    description: "number of exceptions caught"
```
Comment (cijothomas) [inline specification/metrics/new_api.md:498]:
```suggestion

```

description is passed twice.

**Review by cijothomas — APPROVED** [review]
LGTM.
(left few comments about typos)

File: specification/metrics/new_api.md [hunk 5]
```diff
@@ -461,13 +461,77 @@ Example uses for `Histogram`:
...
+[attributes](../common/common.md#attributes) to be passed in as individual
+arguments. The client MAY allow attribute values to be passed in using a more
+efficient way (e.g. strong typed struct allocated on the callstack, tuple). Here
+are some examples that individual language client might consider:
+
+```python
+# Python
+
+http_server_duration.Record(50, {"http.method": "POST", "http.scheme": "https"})
+http_server_duration.Record(100, http_method="GET", http_scheme="http"})
```
Comment (ThomsonTan) [inline specification/metrics/new_api.md:526]: nit: `http_scheme="https"` for consistency with below example?

**Review by jsuereth — APPROVED** [review]

**Review by SergeyKanzhelev — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

**Pull request merged**